src/components/Select/Select.tsx

```diff
@@ -255,6 +262,32 @@ function SelectComponent(
   );
   const hasPrefix = Boolean(prefix);
 
+  const dispatch = useClickTrigger();
+  const handleChange = (e: ChangeEvent<HTMLSelectElement>): void => {
+    const selectedValue = e.target.value;
+    const selectedOption = find(
+      option => option.value === selectedValue,
+      options
+    );
+    const {
+      label: trackingLabel,
+      component = 'select',
+      customParameters
+    } = getOr({}, 'tracking', selectedOption);
+
+    if (trackingLabel) {
+      dispatch({
+        label: trackingLabel,
+        component,
+        customParameters
+      });
+    }
+
+    if (onChange) {
+      onChange(e);
+    }
+  };
```

Comment:
Hm, I am torn about this approach. I would have imagined the tracking prop to be passed to the Select component directly, not as part of each option. This would greatly simplify the code, simplify the usage, and make it consistent with the Checkbox and Selector components. You could send the selected value as a custom parameter.

The approach you chose is a lot more powerful. The question is, do we need this flexibility? Can you think of a use case where the tracking data would be different for each option?